src/filesystem/impls/filer/FilerFileSystem.js

```diff
@@ -316,7 +322,9 @@ define(function (require, exports, module) {
                 return;
             }
 
-            if (options.hasOwnProperty("expectedHash") && options.expectedHash !== stats._hash) {
+            if (_doConsistencyCheck                    &&
```

Comment:
Correct.  We just won't bother checking the hash/cached version against this content, we'll just write it out to disk no matter what.